File: src/main/java/com/proiectPS/demo/model/CryptoTransaction.java
package com.proiectPS.demo.model;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

//todo pretty sure "Transaction" is a SQL keyword
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class CryptoTransaction {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long pk;

    public String coinName;
    public double aquisitionPrice;
    public String userNickname;
    public String aquisitionGMTDate;
    public double amount;
    public String type;

    @Override
    public String toString() {
        return "CryptoTransaction{" +
                "pk=" + pk +
                ", coinName='" + coinName + '\'' +
                ", aquisitionPrice=" + aquisitionPrice +
                ", userNickname='" + userNickname + '\'' +
                ", aquisitionGMTDate='" + aquisitionGMTDate + '\'' +
                ", amount=" + amount +
                ", type='" + type + '\'' +
                '}';
    }
}

File: frontEnd/TudorAngular/src/app/model/crypto-transaction.ts
export class CryptoTransaction {


        public pk : number
        public userNickname : string
        public coinName  :  string;
        public aquisitionPrice : number;
        public amount : number;
        public aquisitionGMTDate : string;
        public type : string
        public currentPrice : number  = 0
        public makingProfit : boolean = false
    
        constructor(currentPrice :  number,type : string,pk : number,coinID : string, coinCurrentPrice : number, amount:  number, currentDate : string, userNickname : string ){
            this.coinName = coinID
            this.aquisitionPrice = coinCurrentPrice
            this.amount = amount
            this.aquisitionGMTDate = currentDate
            this.userNickname = userNickname
            this.pk = pk
            this.type = type
            this.currentPrice = 0
        }
    
}
